src/søknad/steg/2-bosituasjon/Bosituasjon.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { FC, useState } from 'react';
+import { IRoute, Routes } from '../../../routing/Routes';
+
+import { FormattedHTMLMessage, injectIntl, IntlShape } from 'react-intl';
+import Side from '../../../components/side/Side';
+import { useLocation } from 'react-router';
+import { hentNesteRoute } from '../../../routing/utils';
+import { erValgtSvarLiktSomSvar } from '../../../utils/søknad';
+import SeksjonGruppe from '../../../components/gruppe/SeksjonGruppe';
+import LocaleTekst from '../../../language/LocaleTekst';
+import MultiSvarSpørsmål from '../../../components/spørsmål/MultiSvarSpørsmål';
+import { AlertStripeAdvarsel } from 'nav-frontend-alertstriper';
+import { delerSøkerBoligMedAndreVoksne } from './BosituasjonConfig';
+import useSøknadContext from '../../../context/SøknadContext';
+import { IMultiSpørsmål, IMultiSvar } from '../../../models/spørsmal';
+import OmSamboerenDin from './OmSamboerenDin';
+import SøkerSkalFlytteSammenEllerFåSamboer from './SøkerSkalFlytteSammenEllerFåSamboer';
+
+interface Props {
+  intl: IntlShape;
+}
+
+const Bosituasjon: FC<Props> = ({ intl }) => {
+  const { søknad, settSøknad } = useSøknadContext();
+  const { bosituasjon } = søknad;
+  const { søkerDelerBoligMedAndreVoksne } = bosituasjon;
+  const [svarPåHovedspørsmål, settSvarPåHovedspørsmål] = useState('');
+
+  const hovedSpørsmål: IMultiSpørsmål = delerSøkerBoligMedAndreVoksne;
+  const location = useLocation();
+  const nesteRoute: IRoute = hentNesteRoute(Routes, location.pathname);
+  const valgtBosituasjon: string = bosituasjon.søkerDelerBoligMedAndreVoksne
+    .svar_tekst
+    ? bosituasjon.søkerDelerBoligMedAndreVoksne.svar_tekst
+    : '';
+
+  const settSøkerDelerBoligMedAndreVoksne = (svar: string) => {
+    settSøknad({
+      ...søknad,
+      bosituasjon: {
+        ...bosituasjon,
+        søkerDelerBoligMedAndreVoksne: {
+          nøkkel: delerSøkerBoligMedAndreVoksne.spørsmål_id,
+          spørsmål_tekst: intl.formatMessage({
+            id: delerSøkerBoligMedAndreVoksne.tekstid,
+          }),
+          svar_tekst: svar,
+        },
+      },
+    });
+    svarPåHovedspørsmål === '' && settSvarPåHovedspørsmål(svar);
+  };
+
+  const valgtSvar:
+    | IMultiSvar
+    | undefined = hovedSpørsmål.svaralternativer.find((svar) =>
+    erValgtSvarLiktSomSvar(valgtBosituasjon, svar.svar_tekstid, intl)
+  );
+
+  const valgtSvarNøkkel = valgtSvar?.svar_tekstid.split('.')[2];
+
+  const harSøkerSamboerOgLeverIEkteskapsliknendeForhold =
+    valgtSvarNøkkel === 'harEkteskapsliknendeForhold';
+
+  const planerOmÅFlytteSammenEllerFåSamboer =
+    valgtSvarNøkkel === 'borAleneMedBarnEllerGravid' ||
+    valgtSvarNøkkel === 'borMidlertidigFraHverandre' ||
+    valgtSvarNøkkel === 'delerBoligMedAndreVoksne' ||
+    valgtSvarNøkkel === 'tidligereSamboerFortsattRegistrertPåAdresse';
+
```

Comment:
**Info:** 
spørsmålsobjektet (av typen IJaNeiSpørsmål eller IMultiSpørsmål) vi bruker til å sette tekster til spørsmål og svar inneholder tekstid, og ikke selve tekstene. Dette er for å kunne bytte mellom bokmål, nynorsk og engelsk. Tekstidene består av 'bosituasjon.svar.[svarnøkkel]'  - så f.eks. 'bosituasjon.svar.borAleneMedBarnEllerGravid'.  Dette er grunnen til at jeg splitter søkers 'valgteSvar' for så å bare ta nøkkelen på slutten for så å sammenlikne med nøkkelstrengene, for å finne ut om svaret treffer utfallet "planerOmÅFlytteSammenEllerSamboer"-spørsmålet. 

**Diskusjon/Spørsmål:** Er det en bedre måte på å luke ut disse svarsalternativene?  Altså, for å ikke ha så repeterende kode, og i tilfelle noen endrer nøkkel i søknadsobjektet / tekstidene som fører til at vi har flere steder å gjøre endringene på.